### PR TITLE
complete_ingest robot added and some small cleanup

### LINKED ISSUE #14: (IR) re-implement and re-name (old) ingest-cleanup robot
we will rename this robot `complete-ingest`.  Talk to Lynn and group, make sure name agrees in WF object (see https://argo-stage.stanford.edu/view/druid:rv601yh2288) and in other places.

Existing code:
- sdr-preservation-code ingest-cleanup

any code in archive-utils or sdr-replication should either go in preservation_robots or moab-versioning or ... ; 

intend to deprecate:
- sdr-preservation-core
- archive-utils
- sdr-replication

#### from sul-dlss/preservation2017#45

>Current workflow steps  and recommendations (to the extent this analysis can):
>**(9) ingest-cleanup** [code](https://github.com/sul-dlss/sdr-preservation-core/blob/master/lib/sdr_ingest/ingest_cleanup.rb)
>- cleans up the ingest workspaces and signals DOR accessioning that Ingest is completed by >updating an accessioning WF step.
>- **Recommend: no changes needed**

### PR DESCRIPTION
**Pull request by ndushay**
in addition to implementing the complete-ingest robot:

- added placeholder for update-catalog robot (to make it easier to test robots)
- tweaked transfer_object robot:  removed some unnec code
- updated dependencies
- pinned rubocop-rspec to avoid code churn.

Closes #14

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ gem 'honeybadger' # for error reporting / tracking / notifications
 group :development, :test do
   gem 'pry-byebug'
   gem 'rubocop', '~> 0.52.1', require: false # avoid code churn due to rubocop changes
-  gem 'rubocop-rspec'
+  gem 'rubocop-rspec', '~> 1.23.0', require: false # avoid code churn due to rubocop-rspec changes
```
Comment (atz) [inline Gemfile]: Don't think we can legitimately do `require: false` on this or the preceeding line AND expect CI to run rubocop.

File: robots/preservation_ingest/transfer_object.rb [hunk 2]
```diff
@@ -60,19 +60,13 @@ def prepare_deposit_dir
           LyberCore::Log.debug("deposit bag pathname is: #{deposit_bag_pathname}")
           deposit_dir = deposit_bag_pathname.parent
           if deposit_bag_pathname.exist?
-            remove_bag(deposit_bag_pathname)
+            deposit_bag_pathname.rmtree
```
Comment (SaravShah) [inline robots/preservation_ingest/transfer_object.rb]: 👍

**Comment by ndushay** [comment]
@atz  thanks for your comment in Gemfile;  added a commit tweaking it.  Coveralls drop is due to placeholder class without tests.

File: robots/preservation_ingest/complete_ingest.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          LyberCore::Log.error(errmsg)
+          raise(ItemError, errmsg)
+        end
+
+        def update_accession_workflow
+          opts = {
+            elapsed: 1,
+            note: "#{WORKFLOW_NAME} completed on #{Socket.gethostname}"
+          }
+          workflow_service.update_workflow_status('dor', druid, 'accessionWF', 'sdr-ingest-received', 'completed', opts)
```
Comment (SaravShah) [inline robots/preservation_ingest/complete_ingest.rb]: Should we make these constants?

Comment (ndushay) [reply robots/preservation_ingest/complete_ingest.rb]: I dunno.  They are only used once.  What do folks think?

Comment (atz) [reply robots/preservation_ingest/complete_ingest.rb]: Against the use of constants here.  They are params.  That is fine.

File: spec/preservation_ingest/complete_ingest_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      allow(Moab::StorageServices).to receive(:find_storage_object).and_return(mock_so)
+      FileUtils.mkdir_p(deposit_bag_pathname)
+      FileUtils.touch(deposit_bag_pathname + 'bagit_file.txt')
+    end
+    after do
+      deposit_dir_pathname.rmtree if deposit_dir_pathname.exist?
+    end
+
+    it 'raises ItemError if it fails to remove the deposit bag' do
+      expect(deposit_bag_pathname.exist?).to be true
```
Comment (atz) [inline spec/preservation_ingest/complete_ingest_spec.rb]: Not critical, but you can use the `exist` matcher here, if you like:
```ruby
expect(deposit_bag_pathname).to exist
```

Comment (ndushay) [reply spec/preservation_ingest/complete_ingest_spec.rb]: true - will file away for future use.  (xxx? predicates can be the expectation in rspec tests - which I knew but wasn't thinking about)

**Pull request merged**